File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const string = [];
    
    for (let c of s) {
        if (c === "(") {
            string.push("false");
        } else {
            if (!string[0]) return false;
            string.pop();
        }
    }
    
    return string[0] === "false" ? false : true;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = []
    
    for c in s:
        if c == '(':
            answer.append("(")
        else:
            if len(answer) == 0:
                return False
            if answer[-1] == "(":
                answer.pop()

    return True if len(answer) == 0 else False